### PR TITLE
refactor: bonded coins wireframe adjustments

### PR DESCRIPTION
**Pull request by rflechtner**
Combines some changes that result from discussions on the feature branches but are more general than the individual features.

## Checklist:

- [ ] I have verified that the code works
  - [x] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: pallets/pallet-bonded-coins/src/types.rs [hunk 1]
```diff
@@ -36,7 +36,8 @@ impl<LockType> PoolStatus<LockType> {
 
 #[derive(Clone, Encode, Decode, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
 pub struct PoolDetails<AccountId, ParametrizedCurve, Currencies> {
-	pub manager: AccountId,
+	pub owner: AccountId,
+	pub manager: Option<AccountId>,
```
[the remark this answers is not in the document]
Comment (ntn-x2) [reply pallets/pallet-bonded-coins/src/types.rs:40]: Why not make use of Rust's strong type system here? I think it's an option is the right type here.

File: pallets/pallet-bonded-coins/src/types.rs [hunk 2]
```diff
@@ -45,47 +46,46 @@ pub struct PoolDetails<AccountId, ParametrizedCurve, Currencies> {
...
+	pub fn new(owner: AccountId, curve: ParametrizedCurve, bonded_currencies: Currencies, transferable: bool) -> Self {
 		Self {
-			manager,
+			manager: Some(owner.clone()),
+			owner,
 			curve,
 			bonded_currencies,
 			transferable,
-			state,
+			state: PoolStatus::Locked(Default::default()),
```
Comment (ntn-x2) [inline pallets/pallet-bonded-coins/src/types.rs:58]: This should be the `Default` implementation of `PoolStatus`, I'd say.

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 3]
```diff
@@ -115,10 +115,14 @@ pub mod pallet {
 
 		/// The asset id of the collateral currency.
 		type CollateralAssetId: Get<CollateralAssetIdOf<Self>>;
-		/// Who can create new bonded currency pools.
+		/// The origin for most permissionless and priviledged operations.
+		type DefaultOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::AccountId>;
```
Comment (ntn-x2) [inline pallets/pallet-bonded-coins/src/lib.rs:119]: Where would this origin be used? For what extrinsics?

File: pallets/pallet-bonded-coins/src/lib.rs [hunk 4]
```diff
@@ -115,10 +115,14 @@ pub mod pallet {
...
+		/// The dedicated origin for creating new bonded currency pools (typically permissionless).
 		type PoolCreateOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::AccountId>;
+		/// The origin for permissioned operations (force_* transactions).
+		type ForceOrigin: EnsureOrigin<Self::RuntimeOrigin>;
 		/// The type used for pool ids
-		type PoolId: Parameter + MaxEncodedLen + From<[u8; 32]> + Into<Self::AccountId>;
+		type PoolId: Parameter + MaxEncodedLen + From<[u8; 32]> + Into<[u8; 32]>;
```
Comment (ntn-x2) [inline pallets/pallet-bonded-coins/src/lib.rs:126]: `From<[u8; 32]>` is then not necessary. You can create a `[u8; 32]` with this trait and then call `AccountId::from(<[u8; 32]>)`. `PoolId` is not required to deal with `From<[u8; 32]>`, unless I am missing something.

File: pallets/pallet-bonded-coins/src/lib.rs (hunk 3 above)
Comment (rflechtner) [reply pallets/pallet-bonded-coins/src/lib.rs:121]: For most of them, like mint_into, burn_into, start_destroy ... everything that's not creation or a force origin

File: pallets/pallet-bonded-coins/src/lib.rs (hunk 4 above)
Comment (rflechtner) [reply pallets/pallet-bonded-coins/src/lib.rs:128]: I also can't tell you what that was for, but I assumed it had a purpose. @Ad96el why do we need to be able to create a pool id from a 32 byte string?

File: pallets/pallet-bonded-coins/src/types.rs (hunk 2 above)
Comment (rflechtner) [reply pallets/pallet-bonded-coins/src/types.rs:58]: Yeah guess I'll just cherry-pick 9ae15d2310bb6a7f53b8e57cf7e972820f5f674b then. I had it implemented that way, can't tell you when exactly this disappeared.

File: pallets/pallet-bonded-coins/src/lib.rs (hunk 4 above)
Comment (Ad96el) [reply pallets/pallet-bonded-coins/src/lib.rs:128]: I am also not sure anymore. We can get rid of the trait bound, I guess.

File: pallets/pallet-bonded-coins/src/lib.rs (hunk 3 above)
Comment (rflechtner) [reply pallets/pallet-bonded-coins/src/lib.rs:121]: that's what I took from our conversations, that you wanted a configurable origin for all of these things?

File: pallets/pallet-bonded-coins/src/lib.rs (hunk 4 above)
Comment (rflechtner) [reply pallets/pallet-bonded-coins/src/lib.rs:128]: c0722ea8bd2fdc02a1da7ba98e0a852d3069e743

File: pallets/pallet-bonded-coins/src/types.rs (hunk 2 above)
Comment (rflechtner) [reply pallets/pallet-bonded-coins/src/types.rs]: 835cfeacce6711ee77758a5db6e75b59fa690cf2

**Pull request merged**